WordPress/gutenberg
# Update Spacer block minimum width/height

## What?

Prevents Spacer blocks from being zero width or height and sets a practical minimum size that varies according the the unit. Secondarily, some changes propose to share more logic between web and RN and eliminate one inconsistency between the two.

## Why?

* It’s not a spacer unless it takes up space.
* Will also close #47418.
* There is funny margin collapsing that can happen on block-based themes when the size is zero. It's correct in that it acts the same way on the front end but it’s funky in the editor.

## How?

The hook `useCustomUnits` was already in use for the Spacer block. Its return value includes a `step` value for each of the units and that has been coopted for use as the minimum size. The constant that was previously used for all the units was `0` is changed to `1` and only used as the fallback in case `step` does not exist for a unit (very edge case).

## Testing Instructions
1. Insert a Spacer block into the editor.
2. Try sizing it to zero with either the draggable in-canvas size control or the inspector control.
3. Verify that the block will not size below a minimum (that’s dependent on the unit).

### Testing Instructions for Keyboard
1. Tab to the “Height” control in the inspector controls.
2. Enter “0” or use the arrow keys to try changing the value to `0`.
3. Verify that the block will not size below a minimum (that’s dependent on the unit).

## Screenshots or screencast 

### Before

https://user-images.githubusercontent.com/9000376/226230624-8d289e25-7bc0-47eb-a629-825fa9ebcba3.mp4

### After

https://user-images.githubusercontent.com/9000376/226230842-fab4f56c-4794-4e07-958d-ae94b112259f.mp4
